tests/unit/App.spec.ts

```diff
@@ -284,68 +285,56 @@ describe('Incorporation - Define Company page for a BEN (numbered)', () => {
     store.stateModel.business.businessId = ''
 
     // mock user info
-    vi.spyOn((AuthServices as any), 'fetchUserInfo').mockImplementation(() => (
-      {
-        contacts: [
-          {
-            email: 'completing-party@example.com',
-            phone: '123-456-7890'
-          }
-        ],
-        firstname: 'Completing',
-        lastname: 'Party'
-      }
-    ))
+    vi.spyOn((AuthServices as any), 'fetchUserInfo').mockResolvedValue({
+      contacts: [
+        {
+          email: 'completing-party@example.com',
+          phone: '123-456-7890'
+        }
+      ],
+      firstname: 'Completing',
+      lastname: 'Party'
+    })
 
     // mock org info
-    vi.spyOn((AuthServices as any), 'fetchOrgInfo').mockImplementation(() => (
+    vi.spyOn((AuthServices as any), 'fetchOrgInfo').mockResolvedValue({
+      mailingAddress:
       {
-        mailingAddress:
-        {
-          city: 'City',
-          country: 'CA',
-          region: 'BC',
-          postalCode: 'V8V 8V8',
-          street: '1234 Some Street',
-          streetAdditional: 'Suite ABC'
-        }
+        city: 'City',
+        country: 'CA',
+        region: 'BC',
+        postalCode: 'V8V 8V8',
+        street: '1234 Some Street',
+        streetAdditional: 'Suite ABC'
       }
-    ))
+    })
 
     // mock incorporation application
-    vi.spyOn((LegalServices as any), 'fetchFirstOrOnlyFiling').mockImplementation(() => (
-      {
-        header: {
-          name: 'incorporationApplication',
-          filingId: 54321,
-          status: 'DRAFT'
-        },
-        business: {
-          identifier: 'T7654321',
+    vi.spyOn((LegalServices as any), 'fetchFirstOrOnlyFiling').mockResolvedValue({
+      header: {
+        name: 'incorporationApplication',
+        filingId: 54321,
+        status: 'DRAFT'
+      },
+      business: {
+        identifier: 'T7654321',
+        legalType: 'BEN'
+      },
+      incorporationApplication: {
+        nameRequest: {
           legalType: 'BEN'
-        },
-        incorporationApplication: {
-          nameRequest: {
-            legalType: 'BEN'
-          }
         }
       }
-    ))
+    })
 
     // mock filing fees
-    vi.spyOn((PayServices as any), 'fetchFilingFees').mockImplementation(() => (
-      {
-        futureEffectiveFees: 100
-      }
-    ))
+    vi.spyOn((PayServices as any), 'fetchFilingFees').mockResolvedValue({ futureEffectiveFees: 100 })
 
     // mock authorized actions
-    vi.spyOn((LegalServices as any), 'fetchAuthorizedActions').mockImplementation(() => (
-      [...PublicUserActions]
-    ))
+    vi.spyOn((LegalServices as any), 'fetchAuthorizedActions').mockResolvedValue([...PublicUserActions])
 
     // mock GetKeycloakRoles so we don't need a KC token
-    vi.spyOn(utils, 'GetKeycloakRoles').mockImplementation(() => [AuthorizationRoles.PUBLIC_USER])
+    vi.spyOn(utils, 'GetKeycloakRoles').mockReturnValue([AuthorizationRoles.PUBLIC_USER])
```

Comment:
'GetKeycloakRoles' doesn't return a promise (ie, is not async) so here I mock a return value instead of a resolved value.